### ISSUE TITLE
Unable to playback HEVC - Windows

### ISSUE THREAD

**Issue by cmdshft** [brave/brave-browser#40101]
### Description

I run Unifi Protect security cameras, and the cameras recently had an update that allowed the use of the HEVC codec to record with and also view the live feed with. It's great on iOS through their own app, but on the web browser only browsers that support HEVC will be able to play. Brave seems to be unable to play them as such.

### Steps to reproduce

1. Open up Protect Web App
2. Cameras all report that the browser cannot playback the video
3. ???

### Actual result

![image](https://github.com/user-attachments/assets/4cfc5327-8da2-48af-a2a0-a4d0f2f84c65)
![image](https://github.com/user-attachments/assets/012430bd-8011-4acb-a228-536c985688b3)

### Expected result

Should be able to playback HEVC.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

1.68.131

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [X] with Brave Shields disabled
- [X] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by cmdshft** [comment]
Anyone?

**Comment by bsclifton** [comment]
@cmdshft looking into this - seems HEVC is a proprietary codec

I did find this extension you can try - let me know if this works:
https://chromewebstore.google.com/detail/h265-hevc-player/dambgipgbnhmnkdolkljibpcbocimnpd

**Comment by bsclifton** [comment]
And I need to read a bit more 😓 Seems that extension is only for Chrome OS

If you have Chrome installed, I'm curious if the video playback works there?

**Comment by bsclifton** [comment]
@cmdshft I found a solution that works! Special thanks to the guide here:
https://github.com/StaZhu/enable-chromium-hevc-hardware-decoding

Basically, it seems this is only supported if you enable the feature via command line. Not just for Brave, but for any Chromium based browser.

If you're on Windows, there should already be a shortcut for Brave on the desktop. You can edit this and add `--enable-features=PlatformHEVCEncoderSupport` to the end of the `Target` field.
![image](https://github.com/user-attachments/assets/f5cd5406-2fd0-4fec-870c-67395254eda0)

macOS I'd have to look - but I am pretty sure you can do shortcuts with command line arguments too

Let me know if this works! 😄

**Comment by cmdshft** [comment]
@bsclifton Unfortunately, this doesn't seem to work as well. I used the test site, as well as trying the Protect web application and setting my cameras to the HEVC encoder option and still unable to use it.

What's interesting to me is that both Chrome and Edge work without issue and without needing to use this special command on the shortcut for either browser.

**Comment by cmdshft** [comment]
I did manage to resolve this, though. I completely removed Brave (and the browsing data, after backing up passwords to CSV) and reinstalled and it’s working as expected. Not really sure what caused it to break as I have not been using Brave for longer than a few months already. But, it is working none the less.

**Comment by bsclifton** [comment]
@cmdshft super good to hear! 😄 Thank you for digging in and trying some troubleshooting - and then especially thanks for coming back to report what worked for you here.

I'll go ahead and close issue for now. I'll make sure our support team sees this. I am curious why the flag worked for me (I am on Nightly).

**Comment by MrJimmyPenguin** [comment]
I have the exact same issue on popos, still doesnt work. Brave [v 1.68.128]

**Comment by bsclifton** [comment]
cc: @rebron for prioritization

**Comment by MrJimmyPenguin** [comment]
Same issue linux popos and Brave: v 1.68.137

**Comment by 0x1a8510f2** [comment]
Same issue, also with Unifi Protect. I'm on Brave `Version 1.69.168 Chromium: 128.0.6613.138 (Official Build) unknown (64-bit)` installed via Flatpak on openSUSE Tumbleweed on a Framework 13 (AMD Ryzen 7 7840U w/ Radeon 780M Graphics). The aforementioned flag doesn't seem to do anything and neither does starting the browser with a blank `--user-data-dir`.

**Comment by fegauthier** [comment]
Same issue here with Brave on Linux (3080 Ti GPU)

![Image](https://github.com/user-attachments/assets/c35a6e13-3543-4da2-8124-0b07f8fd0a30)

Working well on Windows. Anyone has find out how to fix this?

**Comment by MrJimmyPenguin** [comment]
Still an issue on Brave 173.91

**Comment by difloh** [comment]
I have the same Issue, only with the custom Build Chromium (https://github.com/StaZhu/enable-chromium-hevc-hardware-decoding) i can play HEVC Files. OS: Linux Mint 22. Every other Chromium Browser fails.

**Comment by jwfox5150** [comment]
Bump, still no HEVC support in 1.76.73

**Comment by YShow** [comment]
samue issue trying to see h265 streams on frigate, works fine on the android version but no luck on desktop

**Comment by StefanTT** [comment]
Still not working with brave v1.77.101.
It works with Firefox already. A pity that brave is late to support it.

Test video: https://test-videos.co.uk/vids/bigbuckbunny/mp4/h265/1080/Big_Buck_Bunny_1080_10s_2MB.mp4

**Comment by eygraber** [comment]
There are some magic flags needed on Chrome to get it to work - https://issues.chromium.org/issues/40883357#comment27

For example I needed:

```
google-chrome --ignore-gpu-blocklist --disable-gpu-driver-bug-workaround
    --enable-features=Vulkan,VaapiIgnoreDriverChecks,DefaultANGLEVulkan,VulkanFromANGLE
    --use-gl=angle --use-angle=vulkan --ozone-platform=x11
```

Maybe Brave needs the same.

**Comment by StefanTT** [comment]
@eygraber you are right. With these flags the test video works. I did a bit of testing and with these flags it works for me:
```
brave --enable-features=Vulkan,VaapiIgnoreDriverChecks,DefaultANGLEVulkan,VulkanFromANGLE --ozone-platform=x11 'https://test-videos.co.uk/vids/bigbuckbunny/mp4/h265/1080/Big_Buck_Bunny_1080_10s_2MB.mp4'
```

**Comment by fegauthier** [comment]
> [@eygraber](https://github.com/eygraber) you are right. With these flags the test video works. I did a bit of testing and with these flags it works for me:
> 
> ```
> brave --enable-features=Vulkan,VaapiIgnoreDriverChecks,DefaultANGLEVulkan,VulkanFromANGLE \
>    --ozone-platform=x11 'https://test-videos.co.uk/vids/bigbuckbunny/mp4/h265/1080/Big_Buck_Bunny_1080_10s_2MB.mp4'
> ```

I'm under ZorinOS 17.3 (Which is a fork of Ubuntu). I'm using Brave 1.77.100 and it doesn't work. HEVC videos doesnt work with commands above. I have a 3080 Ti insde my computer.

What knd of setup do you have?

**Comment by StefanTT** [comment]
An AMD graphics card. You tried the full list of arguments that @eygraber posted?

The arguments read as if this is some experimental stuff where VAAPI uses Vulkan for rendering. It is quite possible that this does not yet work with Nvidia drivers.

**Comment by fegauthier** [comment]
Yes, I tried OpenGL instead of Vulkan and it doesn't work either. I red that NVIDIA drivers doesn't support VAAPI. It's weird because with Thorium Browser (Which is a Chromium fork too) it's working. So they did something on the browser side to get it worked.

**Comment by eygraber** [comment]
@fegauthier are you using Wayland or X11? There are different flags depending on which one you're using.

**Comment by fegauthier** [comment]
> [@fegauthier](https://github.com/fegauthier) are you using Wayland or X11? There are different flags depending on which one you're using.

Zorin OS 17.3 is under X11

**Comment by fegauthier** [comment]
@eygraber I changed X11 for Wayland and the issue is the same with other command line

**Comment by eygraber** [comment]
It would be interesting to see what Thorium is doing to enable it to work.

**Comment by fegauthier** [comment]
There is the chrome://gpu report from Thorium

[about-gpu-2025-04-27T12-56-26-323Z.txt](https://github.com/user-attachments/files/19929768/about-gpu-2025-04-27T12-56-26-323Z.txt)

There is the brave://gpu report from Brave

[about-gpu-2025-04-27T12-57-11-253Z.txt](https://github.com/user-attachments/files/19929769/about-gpu-2025-04-27T12-57-11-253Z.txt)

First thing I see is that Video encoding seems to be hardware acelerated on Thorium and not on Brave

**Comment by nathankooistra** [comment]
According to Thorium's github [PATCHES.md](https://github.com/Alex313031/Thorium/blob/main/docs/PATCHES.md)

This is the Patch they used to enable HEVC/H.265: https://github.com/StaZhu/enable-chromium-hevc-hardware-decoding/blob/main/add-hevc-ffmpeg-decoder-parser.patch

**Comment by fegauthier** [comment]
> According to Thorium's github [PATCHES.md](https://github.com/Alex313031/Thorium/blob/main/docs/PATCHES.md) there are a handful of HEVC related Patches listed.
> 
> This is the Patch they used to enable HEVC/H.265 Decoding: https://github.com/StaZhu/enable-chromium-hevc-hardware-decoding/blob/main/add-hevc-ffmpeg-decoder-parser.patch

Would be great if Brave could implement this

**Comment by YShow** [comment]
for anyone still trying, after doing everthing in this thread i found a solution that worked

following this issue plus https://github.com/brave/brave-browser/issues/1024 i found that enabling vaapi with the commands bellow worked for me, the testing video sent by @eygraber also works for me now

these flags in brave: "--enable-features=VaapiVideoDecoder,VaapiVideoEncoder,VaapiVideoDecodeLinuxGL,VaapiIgnoreDriverChecks,PlatformHEVCEncoderSupport --disable-gpu-driver-bug-workaround --skia-graphite-backend"

plus make sure these are enabled

![Image](https://github.com/user-attachments/assets/4c2e35a0-51d0-4066-bc40-2bbed7bf46a6)

**Comment by gyver** [comment]
> **THIS MAY CRASH YOUR BROWSER**
> 
> for anyone still trying, after doing everthing in this thread i found a solution that worked
> 
> following this issue plus [#1024](https://github.com/brave/brave-browser/issues/1024) i found that enabling vaapi with the commands bellow worked for me, the testing video sent by [@eygraber](https://github.com/eygraber) also works for me now
> 
> these flags in brave: "--enable-features=VaapiVideoDecoder,VaapiVideoEncoder,VaapiVideoDecodeLinuxGL,VaapiIgnoreDriverChecks,PlatformHEVCEncoderSupport --disable-gpu-driver-bug-workaround --skia-graphite-backend"
> 
> plus make sure these are enabled
> 
> ![Image](https://github.com/user-attachments/assets/4c2e35a0-51d0-4066-bc40-2bbed7bf46a6)

This doesn't work for me (the media plays without video, no notable error on stdout). Note : I'm using X11 so the Preferred Ozone platform was different from me.

**Comment by YShow** [comment]
> > **THIS MAY CRASH YOUR BROWSER**
> > for anyone still trying, after doing everthing in this thread i found a solution that worked
> > following this issue plus [#1024](https://github.com/brave/brave-browser/issues/1024) i found that enabling vaapi with the commands bellow worked for me, the testing video sent by [@eygraber](https://github.com/eygraber) also works for me now
> > these flags in brave: "--enable-features=VaapiVideoDecoder,VaapiVideoEncoder,VaapiVideoDecodeLinuxGL,VaapiIgnoreDriverChecks,PlatformHEVCEncoderSupport --disable-gpu-driver-bug-workaround --skia-graphite-backend"
> > plus make sure these are enabled
> > ![Image](https://github.com/user-attachments/assets/4c2e35a0-51d0-4066-bc40-2bbed7bf46a6)
> 
> Have you AMD or NVIDIA GPU ?

AMD gpu Vega 56, using opensuse tumbleweed

**Comment by fegauthier** [comment]
> > > **THIS MAY CRASH YOUR BROWSER**
> > > for anyone still trying, after doing everthing in this thread i found a solution that worked
> > > following this issue plus [#1024](https://github.com/brave/brave-browser/issues/1024) i found that enabling vaapi with the commands bellow worked for me, the testing video sent by [@eygraber](https://github.com/eygraber) also works for me now
> > > these flags in brave: "--enable-features=VaapiVideoDecoder,VaapiVideoEncoder,VaapiVideoDecodeLinuxGL,VaapiIgnoreDriverChecks,PlatformHEVCEncoderSupport --disable-gpu-driver-bug-workaround --skia-graphite-backend"
> > > plus make sure these are enabled
> > > ![Image](https://github.com/user-attachments/assets/4c2e35a0-51d0-4066-bc40-2bbed7bf46a6)
> > 
> > 
> > Have you AMD or NVIDIA GPU ?
> 
> AMD gpu Vega 56, using opensuse tumbleweed

It seems to work for AMD gpu but not for NVIDIA gpu

**Comment by popfly510** [comment]
I am also not able to play HEVC videos and have tried all of the recommendations in this thread.


Ubuntu 25.04
Brave v1.79.123
Nvidia RTX 2070 with 570 driver
Wayland display protocol

**Comment by NexGen-3D-Printing** [comment]
I have this issue as well:

Intel 10th Gen
Fedora 42 with KDE Plasma
Intel iGPU

I've install all the media codec's on a system level, all browsers based from FireFox have full HEVC/H265/H264/VP8/VP9 hrdware support, but Brave does not.

Due to this, power consumtion for streaming media is much higher.

The issue does not happen on WIndows, its only on Linux.

**Comment by imanubdesigner** [comment]
Linux Mint 22.1
Brave 1.80.115
RTX 3080 12GB

I've tried ---> sudo apt install chromium-codecs-ffmpeg-extra ffmpeg: no lucky

In Firefox work perfectly!

**Comment by Sameguyy** [comment]
Want to check in 
last ver. Brave
gtx 1080
Arch Wayland Gnome

**Comment by YShow** [comment]
Well my flag solution isnt working anymore even for me :(

**Comment by NexGen-3D-Printing** [comment]
It is working for me now under Bazzite Linux, butr I had to run Brave with all of this junk:

"--enable-features=VaapiVideoDecoder,VaapiVideoEncoder,VaapiVideoDecodeLinuxGL,VaapiIgnoreDriverChecks,PlatformHEVCEncoderSupport --disable-gpu-driver-bug-workaround --skia-graphite-backend"

And I had to enable all of this in the brave://flags section:

<img width="773" height="1347" alt="Image" src="https://github.com/user-attachments/assets/89558b68-00ce-4e85-947a-5df859955d3c" />

brave://gpu report:

<img width="1029" height="653" alt="Image" src="https://github.com/user-attachments/assets/bba9be14-20b5-4e0c-a134-2b68af04401c" />

Tested playback here:

https://lf-tk-sg.ibytedtos.com/obj/tcs-client-sg/resources/video_demo_hevc.html#main-alpha-sample-1

and here:

https://thorium.rocks/misc/h265-tester.html

Hardware h265 seems to be working.

Pretty ridiculous amount of crap one has to go through for the most basic functionality that should just work out of the box.

**Comment by Hyrules** [comment]
This is a thing that need to be implemented. Like others stated it works fine on firefox. I do hope that Brave will give the go soon. I'm also on linux and I can't, as the OP said, watch my security cameras as well. I'm also on unifi.

OS: EndeavourOS Linux x86_64 
Host: MS-7E85 1.0 
Kernel: 6.19.11-arch1-1 
Uptime: 2 hours, 51 mins 
Packages: 1808 (pacman) 
Shell: bash 5.3.9 
Resolution: 3840x2160 
DE: Plasma 6.6.4 
WM: kwin 
Theme: Breeze-Dark [GTK2], Breeze [GTK3] 
Icons: breeze-dark [GTK2/3] 
Terminal: konsole 
CPU: AMD Ryzen 7 9800X3D (16) @ 5.271GHz 
GPU: NVIDIA GeForce RTX 4090 
Memory: 9782MiB / 63668MiB 

Brave v1.88.138

**Comment by NexGen-3D-Printing** [comment]
Did you attempt what I did at all? Just wondering if this works on Arch based OS as well, I have been meaning to check CachyOS, but such steps don't work without additional things being installed and configured in standard Fedora distro's due to some licensing BS.

**Comment by Hyrules** [comment]
Yeah i've tried those flags. They did work at some point but poorly. Last time I tested the flags (about a month ago) they didn't work at all. So I've used firefox to view my cameras.

**Comment by NexGen-3D-Printing** [comment]
I thought that might have been the case, in systems like Bazzite, they have done solid work in implementing the required software for media playback, as that's what the OS is designed more for, I feel your pain though, I actually use Frigate, but with Unifi cameras, Firefox was not a good experience, it gets you by, but Brave is much better, I have several touch screen PC's mounted on walls for viewing and controlling Frigate through Brave WPA's, and they all run Aurora on them with Brave, which is another version from universal blue, this works perfectly for me, but of cause I had to jump through the hoops to get everything to function correctly, but has been perfect for many months now, hopefully they can fix this at some point, as I hate have to dance around a fire in a full moon praying to the tech gods to fix such simple things :)

**Comment by Dawnexa** [comment]
I wonder if Brave will ever implement HEVC support or if we have to wait for Google (and Chromium) to implement Vulkan Video so that we can watch HEVC videos on Linux in Brave...

**Comment by fegauthier-paragon** [comment]
> I wonder if Brave will ever implement HEVC support or if we have to wait for Google (and Chromium) to implement Vulkan Video so that we can watch HEVC videos on Linux in Brave...

I did a PR to support software decoding HEVC with FFMPEG and it's all about HEVC licensing. If they include it in their software they are supposed to pay licensing to HEVC.

So I guess we will have to wait until the implementation of Vulkan by Chromium...

**Comment by Dawnexa** [comment]
> > I wonder if Brave will ever implement HEVC support or if we have to wait for Google (and Chromium) to implement Vulkan Video so that we can watch HEVC videos on Linux in Brave...
> 
> I did a PR to support software decoding HEVC with FFMPEG and it's all about HEVC licensing. If they include it in their software they are supposed to pay licensing to HEVC.
> 
> So I guess we will have to wait until the implementation of Vulkan by Chromium...

Yea I just figured out that they would need to pay licensing to HEVC. This kinda explains why this codec never really took off and why AV1 is becoming more popular...

Lets see how long it takes for Vulkan to arrive...

**Comment by Hyrules** [comment]
Seems that Brave gave the go ahead to finish the PR and merge so this might be coming to users soon. Again thanks for your work on this @fegauthier-paragon

**Comment by m13v** [comment]
this is the chromium hevc licensing gap. unifi flipped the codec default last year and a bunch of property teams couldn't pull live feeds in chrome/brave on linux. on mac it works via videotoolbox, on win it works on intel/amd with hw decode, linux desktops without a patched ffmpeg just see black frames. workaround we used at one site was forcing h.264 stream profile per camera in protect settings, kept recording on hevc but served h.264 to the web client. costs ~30% more storage but every browser plays it.

**Comment by NexGen-3D-Printing** [comment]
> this is the chromium hevc licensing gap. unifi flipped the codec default last year and a bunch of property teams couldn't pull live feeds in chrome/brave on linux. on mac it works via videotoolbox, on win it works on intel/amd with hw decode, linux desktops without a patched ffmpeg just see black frames. workaround we used at one site was forcing h.264 stream profile per camera in protect settings, kept recording on hevc but served h.264 to the web client. costs ~30% more storage but every browser plays it.

I wasn't aware Unifi did this, I have not been able to update my UDM Pro for quite sometime as it requires a full hard reset and I need to rebuild all of my configurations manually, and and I have not had time to take it offline and complete this, so mine is stuck on h265, I could see many people having issues with this and them rolling back the feed to h264, and all of my workarounds to get h265 functioning in Brave, only seem to work easily on an OS like Bazzite or Aurora as they provide these out of the box with full h265 support, where systems like Fedora, Ubuntu etc, cant due to licencing, as stated previously, so the average user, that manages to figure out how to install Brave on Linux, most likely lack the skills or even the time to get everything else functioning just so they can view their camera feeds.

Love to see this fixed for everyone, Brave is about the only close to safe browser that actually works, and is the only one I can recommend normie's switch too, it fixes a lot of safety and targeted ads problems, but then breaks media playback :) 2 step's forwards and 1 step back :)

**Comment by NexGen-3D-Printing** [comment]
Latest flatpak of Brave seems to fully support HEVC decode now on Bazzite and Aurora, but VP9 and AV1 hit the CPU and GPU 3D engines quite hard, so they dont seem to use the hardware decode features from Intel, even though these should be supported on y hardware, I go from 14w at wall usage for a 1080p stream of HEVC to 40w with AV1 and around 25w for VP9, not great on a Laptop, but at least we have HEVC working now without having to pass the following as a command line flag: 

--enable-features=VaapiVideoDecoder,VaapiVideoEncodeLinuxGL,DefaultANGLEVulkan,Vulkan,VulkanFromANGLE --use-angle=vulkan --use-gl=angle

So I will have to continue to force the use of H264 for media consumption, but I now see full support for HEVC using Frigate NVR in Brave, so we are getting somewhere.

For anyone interested, my hardware on this test machine is a 10th Gen Mobile CPU.

**Comment by Hyrules** [comment]
Thing is that the PR for the code has not been pulled yet. I'm following the PR#34796 and it's still open so this is most likely something else. Chromium itself maybe ? On my side the aur version of Brave does not support H.265 as my camera streams are still not showing under it.

**Comment by NexGen-3D-Printing** [comment]
Its definately working under ublue-OS, tested today as I did a few reinstalls on several Latitue 7210 2 in 1's and I specifically test H264/H265/VP9/AV1 monitored the encode/decode and 3D and H265 is hitting the media decoder now, and no launch options were set, VP9 and AV1 are hitting the GPU 3D engine only and spiking power consumption.

**Comment by NexGen-3D-Printing** [comment]
Tested Brave and Brave Origin under Fedora KDE 44, hardware decode for HEVC is fully functional now on a default install, no launch flags required, but you need to enable Vulkan in brave://flags as its disabled by default.

Note, I have enabled non-free media codec support under Fedora as this is not included under a default install, this script makes things like this very easy: https://github.com/k-mktr/fedora-things-to-do.

<img width="622" height="423" alt="Image" src="https://github.com/user-attachments/assets/027e1321-cb0c-46b2-bb65-24ed3d190def" />

<img width="779" height="330" alt="Image" src="https://github.com/user-attachments/assets/23f662b5-fc0a-420e-8952-f647fc29fff7" />

Highly recommend trying Brave Origin.

**Comment by jstebenne** [comment]
I just tested on Brave 1.90.121 with Ubuntu 24.04, and it is **NOT WORKING**. @NexGen-3D-Printing Whatever improvement you are seeing is unrelated to Brave. The fix in the Brave codebase is not even merged, even though the [pull request](https://github.com/brave/brave-core/pull/34796) is moving in the right direction.

**Comment by NexGen-3D-Printing** [comment]
Maybe its a Fedora thing, but Bazzite, Aurora, Fedora along with CachyOS all support full HEVC hardware decoding in Brave, have tested with thefollowing two sites:

https://lf-tk-sg.ibytedtos.com/obj/tcs-client-sg/resources/video_demo_hevc.html#main-alpha-sample-1

https://thorium.rocks/misc/h265-tester.html

Also tested with Plex through Brave and running direct stream HEVC encoded video, and I see a spike in the hardware encode/decode engine when playing any H264 or H265 video, along with Frigate NVR, but VP8/9 and AV1, only hit the GPU 3D Engine, so I dont think these are fully hardware supported, but none hit the CPU at all.

Both test machines are 10th Gen Intel Mobile Chips.

Tested, Flatpak and native versions of Brave and the native version of Brave Origin Beta (No Flatpak available) - Version 1.91.143

What does brave://gpu show under Ubuntu? Is the version under Ubuntu a Snap package?

**Comment by Hyrules** [comment]
Like jstebenne said. This has not been pulled yet in the Brave code. I'm on Arch / Endeavour OS and i'm not getting any HEVC x.265 decoding in Brave although Firefox works fine. This is possibly related to an Nvidia. If you are on a AMD video card or intel it might already work.

Some distros seems to do something some other don't like install some kind of codecs. Those codecs needs some kind of licences AFAIK. I'm on a 9800X3D with 64 GB of RAM and a 4090. I'm just curious as what those distros have that ours don't. It might be working for you but it's not because of Brave unfortunately. I'm also following the PR. 

Cachy has this package in it's extra : https://packages.cachyos.org/package/cachyos-extra-v4/x86_64_v4/x265

That said, Vulkan is also not a thing in chrome base browsers at the moment on nvidia because of this :  

https://issues.chromium.org/issues/324003973

https://issues.chromium.org/issues/324003973#comment30

Nvidia has offered to make the browser compatible with vulkan but chromium guys does not seem to be in any urge to push this.

Also I wouldn't use Thorium as it is unfrequently updated but it does pull the same fix as fegauthier-paragon is doing in his PR.

**Comment by NexGen-3D-Printing** [comment]
Im using Intel 10th gen eGPU's I have a number of AMD machines as well, so I can test those, but I dont use nVidia gear at all, as its not a great hardware choice for Linux, I assume the issues you have are around that, Vulkan is working under my Intel GPU's, and I would assume the same for AMD.

You are most likely correct, and the fix is not from Brave, but the only thing I was concerned about, was HW decode for HEVC in PWA's, and I now have that, Brave installs with the flag hardware-decode enabled by default, if I toggle this off, then it shows as Software HEVC Decode support, if I set to default, it will show Hardware HEVC Decode.

I also would not use Thorium either, I only linked to that test video, as its an easy way to test if your browser supports HEVC decoding, I'm 100% Brave across all platforms, and now Brave Origin, except Android, I use Firefox Nightly as it has full support for extensions, where all the other browsers dont, including Brave (Excepetion was Kiwi, but that project is dead now)

**Comment by Dawnexa** [comment]
> Im using Intel 10th gen eGPU's I have a number of AMD machines as well, so I can test those, but I dont use nVidia gear at all, as its not a great hardware choice for Linux, I assume the issues you have are around that, Vulkan is working under my Intel GPU's, and I would assume the same for AMD.

As far as I'm aware NVIDIA doesn't support VA-API hardware decode for Linux. 

Quote from the Pull Request:

> Chromium's VA-API hardware decode pipeline is fundamentally incompatible with NVIDIA GPUs on Linux

**Comment by NexGen-3D-Printing** [comment]
> > Im using Intel 10th gen eGPU's I have a number of AMD machines as well, so I can test those, but I dont use nVidia gear at all, as its not a great hardware choice for Linux, I assume the issues you have are around that, Vulkan is working under my Intel GPU's, and I would assume the same for AMD.
> 
> As far as I'm aware NVIDIA doesn't support VA-API hardware decode for Linux.
> 
> Quote from the Pull Request:
> 
> > Chromium's VA-API hardware decode pipeline is fundamentally incompatible with NVIDIA GPUs on Linux

The support under Linux for nVidia is one of the many reasons I churned away from nVidia products after the 10 Series, life under Linux is a lot easier with Intel and AMD, not slamming the nVidia products, we have what we have, but Linux is a lot better experience under Intel or AMD, or both at the same time.

**Comment by Hyrules** [comment]
I have been using Nvidia as my video card for the past I don't remember when years. On linux things are fine for me. 99.9% of my stuff is working properly. At the moment this is the only issue I have with my pc and Nvidia otherwise my applications and Games are working as expected and I would dare to say very well. I don't have any kind of instability. Wayland is working as expected, plasmashell is stable. The idea that nvidia is unstable under linux and causes issues is an old myth at this point in time. I see a lot of people with issue with both Amd and Nvidia cards. This might have been an issue in the past but in the recent years thing are working normally.

That said, there is an nvidia va-api driver that exists and still in development by a single dev :

https://github.com/elFarto/nvidia-vaapi-driver

Still a lot to do but some stuff is starting to work. Don't believe what is written on the front page that chromium based browser does not work it's not true anymore. The dev didn't update the page to document this. That said you still need to add the flags to start Brave and for some stuff it's clearly stated what will never work.

**Comment by jstebenne** [comment]
Thanks a lot to everyone involved in completing the [PR](https://github.com/brave/brave-core/pull/34796)!! 

If I understand the [release schedule](https://github.com/brave/brave-browser/wiki/Brave-Release-Schedule) correctly, HEVC support will be included in the 1.93.x release which is **scheduled to be released on July 28, 2026**.

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.93.118 Chromium: 150.0.7871.114 (Official Build) beta (64-bit)
-- | --
Revision | 637ddd379d39c78e83c74d9a6ce8cf7459b8f5c5
OS | Windows 11 Version 25H2 (Build 26200.8655)

- Verified steps from [brave-core#34796](https://github.com/brave/brave-core/pull/34796)
- Verified able to play H.265/HEVC+AAC audio-video files
- Verified `FFmpegVideoDecoder` is used for HEVC content 
- Verified running `MediaSource.isTypeSupported('video/mp4;codecs="hev1.1.6.L93.B0"')` on console shows `true` for both H.264/H.264 files

https://github.com/user-attachments/assets/fd6850de-3120-49c9-bd8c-8f84cdd92d2d

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.93.120 Chromium: 150.0.7871.125 (Official Build) beta (64-bit) 
-- | --
Revision | 3de17fad2596dc87a5da07e9505a22b75ff1b52c
OS | Linux

- Verified steps from [brave-core#34796](https://github.com/brave/brave-core/pull/34796)
- Verified able to play H.265/HEVC+AAC audio-video files
- Verified `FFmpegVideoDecoder` is used for HEVC content 
- Verified running `MediaSource.isTypeSupported('video/mp4;codecs="hev1.1.6.L93.B0"')` on console shows `true` for both H.264/H.264 files

https://github.com/user-attachments/assets/90ee8420-7fde-45fb-be69-86523ad3bb00

**Comment by MrJimmyPenguin** [comment]
This is still an issue on [Brave 1.93.138 (Official Build) (x86_64)]

**Comment by Hyrules** [comment]
It was available for a while and they moved it under a flag which is disabled by default : 

brave://flags/#brave-ffmpeg-software-hevc-decoder

I think this causes some issues for some people in video playback. I personnaly don't have any problem with it and it works fine.

**Comment by jstebenne** [comment]
Thanks for sharing the flag, I was wondering why it didn't work the other day but didn't have time to investigate.

I do have performance issues with it. Videos with AV1 enabled in Unifi Protect take a while to load, and streams are very slow, like 1 frame every 5 seconds.

@Hyrules Do you have an issue ref ID you can share where I can follow progress on that?

**Comment by Hyrules** [comment]
no. I think I saw some in the commit they did about it but I didn't went any further. For me UNVR / unifi protect works fine with the flag enabled. I don't see an issue with the framerate and the streams load normally.
